frontend-react/src/components/Admin/AdminReceiverDashboard.tsx

```diff
@@ -457,56 +500,165 @@ function MainRender(props: {
+                data-orgrecvname={combinedName}
+                data-allslicesfiltered={allSlicesFiltered}
+            >
+                <Grid className={`title-column ${titleClassName}`}>
+                    <div className={"title-text"}>
+                        {orgName}
+                        <br />
+                        {recvrName}
+                        <br />
+                        {successRate}%
+                    </div>
                 </Grid>
-            );
-        }
+                <ScrollSyncPane enabled>
+                    <Grid row className={"horizontal-scroll"}>
+                        <Grid row className={"week-column"}>
+                            {perDayElements}
+                        </Grid>
+                    </Grid>
+                </ScrollSyncPane>
+            </Grid>
+        );
         perDayElements = [];
+        visibleSliceCount = 0;
         // </editor-fold>
     } // while
+    return perReceiverRowElements;
+}
+
+/**
+ * Some filtering doesn't need to modify rows, just show/hide rows.
+ * This function loops over "pre-rendered" rows and decide which to
+ * include.
+ */
+function FilterRenderedRows(props: {
+    renderedRows: JSX.Element[];
+    filterRowStatus: SuccessRate;
+    filterRowReceiver: string;
+    filterErrorText: string;
+    onClick: (dataItem: AdmConnStatusDataType[]) => void;
+}) {
+    const filterRowStatus = props.filterRowStatus.toUpperCase(); // enum is uppercased
+    const filterRowReceiver = props.filterRowReceiver.toLowerCase();
+    const filterErrorText = props.filterErrorText.trim().toUpperCase();
+    const renderedRows = props.renderedRows;
+
+    const resultArray: JSX.Element[] = [];
+    for (let offset = 0; offset < renderedRows.length; offset++) {
+        const renderedRow = renderedRows[offset];
+        const rowStatus = renderedRow.props["data-rowstatus"] || "";
+        const orgRecvName = renderedRow.props["data-orgrecvname"] || "";
+        const allSlicesFilteredOut =
+            renderedRow.props["data-allslicesfiltered"] || false;
+
+        // filters remove rows.
+        const hideRowBasedOnStatus =
+            filterRowStatus !== SuccessRate.UNDEFINED
+                ? rowStatus !== filterRowStatus
+                : false;
+
+        const hideRowBasedOnRecv =
+            filterRowReceiver !== ""
+                ? !orgRecvName.includes(filterRowReceiver)
+                : false;
+
+        // this filter removed ALL slices from the day, so hide the row
+        const hideRowBasedOnErrorFilter =
+            filterErrorText !== "" ? allSlicesFilteredOut : false;
+
+        const hideRow =
+            hideRowBasedOnStatus ||
+            hideRowBasedOnRecv ||
+            hideRowBasedOnErrorFilter;
+
+        if (!hideRow) {
+            resultArray.push(renderedRow);
+        }
+    }
+
+    if (resultArray.length === 0) {
+        return <>No rows matching time range and any filters</>;
```

Comment:
[nit] makes more sense to me for this to be a vanilla JS fn that returns data, and then you just wrap that in jsx in the render. I know that would mean refactoring along the lines I mentioned elsewhere - doing the filter before dealing with JSX, so feel free to ignore

Reply:
Excellent point. I think I made it a component initially as I refactored just move stuff around and never noticed it was really just plain js. Now that you mentioned it, it's going to bother me, so I'll fix.